src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php

```diff
@@ -1212,6 +1290,38 @@ private function getErrorMessages()
                     'Admin.Notifications.Error'
                 ),
             ],
+            CannotEditCmsPageException::class => [
+                CannotEditCmsPageException::INSUFFICIENT_PERMISSION_ALL_SHOPS => $this->trans(
+                    'You don\'t have permissions to edit this page in all shops it belongs to.',
+                    'Admin.Cms.Notification'
+                ),
+                CannotEditCmsPageException::INSUFFICIENT_PERMISSION_ASSOCIATIONS => $this->trans(
+                    'You can only change the association for shops you have permissions for.',
+                    'Admin.Cms.Notification'
+                ),
+            ],
         ];
     }
+
+    protected function employeeHasAccessAssociatedShops($cmsPageId)
```

Comment:
```suggestion
    protected function employeeHasAccessAssociatedShops(int $cmsPageId): bool
```

Reply:
I think this function should be defined somewhere else, like in parent `FrameworkBundleAdminController`, because this is a frequent thing to test, to which in Prestashop is not payed enough attention.
In fact, an employee, that has not access to all shops in multishop context, should NEVER be able to change shop associations for a shop he does not belong to.